pyccel/ast/utilities.py

```diff
@@ -312,15 +341,30 @@
 
     elif isinstance(expr, IndexedElement):
         base = expr.base
+        rank = base.rank
+
+        # If pos indexes base then recurse
+        base_container_rank = base.class_type.container_rank
+        if -pos < rank-base_container_rank:
+            return insert_index(base, pos+base_container_rank, index_var)
```

Comment:
/bot accept
This code seems logical but there is no use case currently. It could be replaced with an assertion but something should enforce this condition for readability.